src/api/app/services/monitor_controller_service/building_information_updater.rb

```diff
@@ -0,0 +1,50 @@
+module MonitorControllerService
+  class BuildingInformationUpdater
+    attr_accessor :workers
+
+    def initialize
+      @workers = {}
+    end
+
+    def call
+      initialize_workers
+      update_workers
+      self
+    end
+
+    private
+
+    def worker_status
+      @worker_status ||= Xmlhash.parse(WorkerStatus.hidden.to_xml)
+    end
+
+    def initialize_workers
+      @workers = Hash[worker_status.elements('idle').collect { |b| [worker_id(b), {}] }]
+    end
+
+    def update_workers
+      @workers.merge!(Hash[@worker_status.elements('building').collect { |b| [worker_id(b), workers_hash(b, calculate_delta(b['starttime'].to_i))] }])
+    end
+
+    def workers_hash(b, delta)
+      Hash[['project', 'repository', 'package', 'arch', 'starttime'].collect { |k| [k, b[k]] }].merge('delta' => delta)
+    end
+
+    def calculate_delta(starttime)
+      delta = (Time.now - Time.at(starttime)).round
+      delta = 5 if delta < 5
+      delta = max_time if delta > max_time
+      delta = (100 * Math.sin(Math.acos(1 - (Float(delta) / max_time)))).round
+      delta = 100 if delta > 100
+      delta
```

Comment:
something that I noticed is that `delta` is the only value in the xml which isn't encoded as string https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/monitor_controller.rb#L55, we should probably change it. What do you think @hennevogel ?

Reply:
Hm, everything in XML is a string right?

Reply:
yes, i thought so :) but I'm not sure if I change I will break someone consuming it

Reply:
types can't be transferred over the wire.